k8s.mk

```diff
@@ -0,0 +1,116 @@
+
+ifeq (,$(shell which kubectl))
+ifeq (,$(shell which oc))
+$(error oc or kubectl is required to proceed)
+else
+K8S_CLI := oc
+endif
+else
+K8S_CLI := kubectl
+endif
+
+ifeq ($(shell $(K8S_CLI) api-resources --api-group='route.openshift.io'  2>&1 | grep -o routes),routes)
+PLATFORM := openshift
+else
+PLATFORM := kubernetes
+endif
+
+# minikube handling
+ifeq ($(shell $(K8S_CLI) config current-context 2>&1),minikube)
+# check ingress addon is enabled
+ifeq ($(shell minikube addons list -o json | jq -r .ingress.Status), disabled)
+$(error ingress addon should be enabled on top of minikube)
+endif
+export ROUTING_SUFFIX := $(shell minikube ip).nip.io
+endif
+
+_create_namespace:
+	$(K8S_CLI) create namespace $(NAMESPACE) || true
+
+_gen_configuration_env:
+	mkdir -p $(INTERNAL_TMP_DIR)
+	echo "export RELATED_IMAGE_devworkspace_webhook_server=$(DWO_IMG)" > $(RELATED_IMAGES_FILE)
+ifeq ($(PLATFORM),kubernetes)
+	echo "export WEBHOOK_SECRET_NAME=devworkspace-operator-webhook-cert" >> $(RELATED_IMAGES_FILE)
+endif
+	cat ./deploy/templates/components/manager/manager.yaml \
+		| yq -r \
+			'.spec.template.spec.containers[]?.env[] | select(.name | startswith("RELATED_IMAGE")) | "export \(.name)=\"$${\(.name):-\(.value)}\""' \
+		>> $(RELATED_IMAGES_FILE)
+	cat $(RELATED_IMAGES_FILE)
+
+### install: Install controller in the configured Kubernetes cluster in ~/.kube/config
+install: _check_cert_manager _print_vars _init_devworkspace_crds _create_namespace generate_deployment
```

Comment:
it's not only for K8s but also for OpenShift. Maybe then better name file `deploy.mk`?

Reply:
Makes sense - I've renamed it to deploy.mk!